Make cartographer compatible with ROS2.

clalancette opened the pull request:
We have to disable all of the tests here because they all require
either gtest or gmock, neither of which is vendored cartographer.
Because of this, you would have to install a system
version of both of them, but that causes problems with all
other tests in the system.  Specifically, what happens is
that the system versions are (most likely) compiled with
different flags from the vendored versions in other tests.
When attempting to run the tests, the dynamic linker first
looks in the system places, finds the .so, and then doesn't
bother with the built one.  The tests typically crash in
bizarre ways after that.  For now, workaround all of this
mess just by disabling the tests here.

Signed-off-by: Chris Lalancette <clalancette@osrfoundation.org>

connects to ros2/ros2#342

CI:
* TB2 Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_turtlebot-demo_linux&build=20)](http://ci.ros2.org/job/ci_turtlebot-demo_linux/20/)
* TB2 Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_turtlebot-demo_linux-aarch64&build=36)](http://ci.ros2.org/job/ci_turtlebot-demo_linux-aarch64/36/)

mikaelarguedas commented:
Same remarks as for ceres-solver, please update the maintainer tags, update dependency list on CI and follow the forking guidelines. Otherwise looks good to me

clalancette commented:
I fixed up the repository to enable issues and change the default branch.  I also updated the maintainer tags.  As far as the dependencies go, they are all pulled in either directly or indirectly, so I think we are good to go with those.

mikaelarguedas commented on package.xml:
```diff
@@ -30,14 +30,12 @@
 
   <depend>boost</depend>
-  <depend>ceres-solver</depend>
+  <depend>ceres_solver</depend>
```
this change was actually not necessary we just needed to install the ceres-solver dependency from apt
